src/fprime_gds/common/fpy/types.py

```diff
@@ -0,0 +1,86 @@
+from dataclasses import dataclass
+from enum import Enum
+import struct
+from fprime.common.models.serialize.type_base import BaseType
+from fprime.common.models.serialize.time_type import TimeType
+from fprime.common.models.serialize.numerical_types import (
+    F32Type,
+    F64Type,
+    I8Type,
+    I16Type,
+    I32Type,
+    I64Type,
+    U8Type,
+    U16Type,
+    U32Type,
+    U64Type,
+)
+
+
+class StatementType(Enum):
+    DIRECTIVE = 0
+    CMD = 1
+
+
+@dataclass
+class StatementTemplate:
+    statement_type: StatementType
+    opcode: int
+    name: str
+    args: list[type[BaseType]]
```

Comment:
As you pointed out during the review, I think ValueType is what we're looking for here and in other uses of BaseType in this code. Represents a value and needs a constructor